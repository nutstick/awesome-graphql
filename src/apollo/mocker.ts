import { createMockerFromIntrospection } from 'fraql/mock';
import introspectionData from '../../schema.json';

export default createMockerFromIntrospection(introspectionData, {
  mocks: {
    Url: () => 'My article title',
  },
});
