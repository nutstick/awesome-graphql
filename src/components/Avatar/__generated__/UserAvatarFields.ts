/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL fragment: UserAvatarFields
// ====================================================

export interface UserAvatarFields {
  __typename: "User";
  avatar: string;
  username: string;
}
