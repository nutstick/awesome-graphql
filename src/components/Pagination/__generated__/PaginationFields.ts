/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL fragment: PaginationFields
// ====================================================

export interface PaginationFields {
  __typename: "PostConnection" | "CommentConnection" | "UserConnection";
  totalCount: number;
}
