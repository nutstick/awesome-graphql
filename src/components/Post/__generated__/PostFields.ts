/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL fragment: PostFields
// ====================================================

export interface PostFields_author {
  __typename: "User";
  avatar: string;
  username: string;
}

export interface PostFields_comments_author {
  __typename: "User";
  avatar: string;
}

export interface PostFields_comments {
  __typename: "Comment";
  id: string;
  author: PostFields_comments_author;
  content: string;
}

export interface PostFields {
  __typename: "Post";
  author: PostFields_author;
  image: any | null;
  caption: string | null;
  comments: PostFields_comments[] | null;
}
