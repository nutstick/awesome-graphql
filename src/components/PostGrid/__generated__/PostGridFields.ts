/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL fragment: PostGridFields
// ====================================================

export interface PostGridFields_edges_node_author {
  __typename: "User";
  id: string;
  avatar: string;
  username: string;
}

export interface PostGridFields_edges_node_comments {
  __typename: "CommentConnection";
  totalCount: number;
}

export interface PostGridFields_edges_node_likes {
  __typename: "UserConnection";
  totalCount: number;
}

export interface PostGridFields_edges_node {
  __typename: "Post";
  id: string;
  author: PostGridFields_edges_node_author;
  image: any | null;
  caption: string | null;
  comments: PostGridFields_edges_node_comments | null;
  likes: PostGridFields_edges_node_likes | null;
}

export interface PostGridFields_edges {
  __typename: "PostEdges";
  node: PostGridFields_edges_node;
  cursor: string;
}

export interface PostGridFields {
  __typename: "PostConnection";
  edges: PostGridFields_edges[] | null;
}
