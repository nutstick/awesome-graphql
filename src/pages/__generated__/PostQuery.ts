/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL query operation: PostQuery
// ====================================================

export interface PostQuery_posts_edges_node_author {
  __typename: "User";
  id: string;
  avatar: string;
  username: string;
}

export interface PostQuery_posts_edges_node_comments_author {
  __typename: "User";
  avatar: string;
}

export interface PostQuery_posts_edges_node_comments {
  __typename: "Comment";
  id: string;
  author: PostQuery_posts_edges_node_comments_author;
  content: string;
}

export interface PostQuery_posts_edges_node {
  __typename: "Post";
  id: string;
  author: PostQuery_posts_edges_node_author;
  image: any | null;
  caption: string | null;
  comments: PostQuery_posts_edges_node_comments[] | null;
}

export interface PostQuery_posts_edges {
  __typename: "PostEdges";
  node: PostQuery_posts_edges_node;
}

export interface PostQuery_posts {
  __typename: "PostConnection";
  edges: PostQuery_posts_edges[] | null;
}

export interface PostQuery {
  posts: PostQuery_posts | null;
}
