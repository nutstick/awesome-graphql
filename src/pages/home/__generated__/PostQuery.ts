/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL query operation: PostQuery
// ====================================================

export interface PostQuery_posts_edges_node_author {
  __typename: "User";
  id: string;
  avatar: string;
  username: string;
}

export interface PostQuery_posts_edges_node_comments {
  __typename: "CommentConnection";
  totalCount: number;
}

export interface PostQuery_posts_edges_node_likes {
  __typename: "UserConnection";
  totalCount: number;
}

export interface PostQuery_posts_edges_node {
  __typename: "Post";
  id: string;
  author: PostQuery_posts_edges_node_author;
  image: any | null;
  caption: string | null;
  comments: PostQuery_posts_edges_node_comments | null;
  likes: PostQuery_posts_edges_node_likes | null;
}

export interface PostQuery_posts_edges {
  __typename: "PostEdges";
  node: PostQuery_posts_edges_node;
  cursor: string;
}

export interface PostQuery_posts {
  __typename: "PostConnection";
  edges: PostQuery_posts_edges[] | null;
  totalCount: number;
}

export interface PostQuery {
  posts: PostQuery_posts | null;
}
