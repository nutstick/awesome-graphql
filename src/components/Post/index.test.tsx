import React from 'react';
import renderer from 'react-test-renderer';
import mocker from '../../apollo/mocker';
import { Post } from './index';
import { PostFields } from './__generated__/PostFields';

describe('Post', () => {
  it('render', () => {
    const data = mocker.mockFragment<PostFields>(Post.fragments.fields);

    const component = renderer.create(<Post {...data} index={0} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
