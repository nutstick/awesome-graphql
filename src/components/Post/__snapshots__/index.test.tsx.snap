// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post render 1`] = `
<div
  className="ant-card ant-card-bordered ant-card-hoverable"
  style={
    Object {
      "width": 240,
    }
  }
>
  <div
    className="ant-card-cover"
  >
    <img
      alt="example"
      src="My article title"
    />
  </div>
  <div
    className="ant-card-body"
    style={Object {}}
  >
    <div
      className="ant-card-meta"
    >
      <div
        className="ant-card-meta-avatar"
      >
        <span
          className="ant-avatar ant-avatar-circle ant-avatar-image"
          style={Object {}}
        >
          <img
            onError={[Function]}
            src="Hello World"
          />
        </span>
      </div>
      <div
        className="ant-card-meta-detail"
      >
        <div
          className="ant-card-meta-description"
        >
          Hello World
        </div>
      </div>
    </div>
  </div>
</div>
`;
